### ISSUE TITLE
Make kernel logs accessible for seed machines.

### ISSUE THREAD

**Issue by hardikdr** [gardener/gardener#2991]
**How to categorize this issue?**

/area logging
/area ops-productivity
/kind enhancement
/priority normal

**What would you like to be added**: Currently MCM replaces the unhealthy machines after a default timeout of 10mins. A machine is considered unhealthy when either of the following node-conditions sets to True `KernelDeadlock`, `FilesystemReadOnly`, `DiskPressure`, or `KubeletReady` to false.

Lately, most of the time, we have been seeing machines being replaced due to the `Kubelet stopped posting status` issue, and rarely any other node-conditions. This doesn't tell us much about the actual root-cause, and as the machine is already gone, we also don't have much to investigate on. 

We must know why the kubelet or other components stops working, specifically/at-least for the seed worker-machines, as they host the critical workload. Currently, Grafana in seed machines does expose kubelet logs, but it might not always be sufficient if the process was abruptly killed for some reason. 

Opening this issue to discuss if we can also fetch more logs from the seed-machines.
- For example, [NPD](https://github.com/kubernetes/node-problem-detector/blob/master/config/kernel-monitor.json) simply watches the `/dev/kmsg` to set the node-conditions. We could store these logs for our references. 

**Why is this needed**: For better observability of the seed machines, and if there is a scope of improvement from MCM or cloud-provider.

**Comment by Kristian-ZH** [comment]
Sounds reasonable, we will do this.

Thanks for the issue :)

**Comment by hardikdr** [comment]
thanks @Kristian-ZH .

@vlvasilev mentioned there is also a plugin for the kernel-logs in fluent-bit. 
I see this, https://docs.fluentbit.io/manual/pipeline/inputs/kernel-logs , probably useful.

**Comment by vlvasilev** [comment]
I did some investigation and tests and it seems that the fluent-bit kmsg plugin does not work without mounting the device /dev/kmsg. As I see the only way to mount a device is to use the [Kubernetes Device Plugin](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/) which we have to write on our own and deploy it as DaemonSet. I have tried to use the 'tail' plugin but without any result. It looks like the 'tail' plugin does not read from /dev directory.
